File: 351003/Matelenok/publisher/src/main/java/by/distcomp/app/dto/UserRequestTo.java
package by.distcomp.app.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;

public record UserRequestTo(
        @NotBlank
        @Size(min = 2, max = 64)
        String login,
        @NotBlank
        @Size(min = 8, max = 128)
        String password,
        @NotBlank
        @Size(min = 2, max = 64)
        String firstname,
        @NotBlank
        @Size(min = 2, max = 64)
        String lastname,

        @Size(min = 2, max = 64)
        String role
) { }

File: 351003/brezgunov/distcomp/src/main/kotlin/com/distcomp/dto/user/UserRequestTo.kt
package com.distcomp.dto.user

import com.distcomp.entity.Role
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class UserRequestTo(
    val id: Long? = null,
    @field:NotBlank
    @field:Size(min = 2, max = 64)
    val login: String,
    @field:NotBlank
    @field:Size(min = 8, max = 128)
    val password: String,
    @field:NotBlank
    @field:Size(min = 2, max = 64)
    val firstname: String,
    @field:NotBlank
    @field:Size(min = 2, max = 64)
    val lastname: String,
    @field:NotNull
    var role: Role = Role.CUSTOMER
)
